### PR TITLE
Extend M421 with I and J parameters

### PR DESCRIPTION
**Pull request by AnHardt**
Extend M421 with I and J parameters to access its mesh-points directly.

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -5906,27 +5906,40 @@ inline void gcode_M410() { stepper.quick_stop(); }
...
+    if ((hasJ = code_seen('J'))) j = code_value();
     if ((hasZ = code_seen('Z'))) z = code_value();
 
     if (hasX && hasY && hasZ) {
 
       int8_t ix = mbl.select_x_index(x),
              iy = mbl.select_y_index(y);
 
-      if (ix >= 0 && iy >= 0)
+      if (ix >= 0 && ix < MESH_NUM_X_POINTS && iy >= 0 && iy < MESH_NUM_Y_POINTS)
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: `select_x_index` and `select_y_index` can never return values >= to `MESH_NUM_X_POINTS` or `MESH_NUM_Y_POINTS`. If a point is too far out of range, these functions return a `-1`.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -5906,27 +5906,40 @@ inline void gcode_M410() { stepper.quick_stop(); }
...
         mbl.set_z(ix, iy, z);
       else {
         SERIAL_ERROR_START;
         SERIAL_ERRORLNPGM(MSG_ERR_MESH_XY);
       }
     }
-    else {
+    else if (hasI && hasJ && hasZ) {
+      if (i >= 0 && i < MESH_NUM_X_POINTS && y >= 0 && j < MESH_NUM_Y_POINTS)
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Oops. The `y` in this line should be `j`.

**Comment by AnHardt** [comment]
Updated.

File: Marlin/Marlin_main.cpp [hunk 3]
```diff
@@ -5919,7 +5923,16 @@ inline void gcode_M410() { stepper.quick_stop(); }
...
+        mbl.set_z(i, j, z);
+      else {
+        SERIAL_ERROR_START;
+        SERIAL_ERRORLNPGM(MSG_ERR_MESH_XY);
+      }
+    }
+    else 
+    {
       SERIAL_ERROR_START;
       SERIAL_ERRORLNPGM(MSG_ERR_M421_REQUIRES_XYZ);
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: I guess we'll need to fix this error too, so it mentions I and J parameters.

File: Marlin/Marlin_main.cpp [hunk 4]
```diff
@@ -5899,12 +5899,16 @@ inline void gcode_M410() { stepper.quick_stop(); }
 
   /**
    * M421: Set a single Mesh Bed Leveling Z coordinate
+   * Eter with 'M421 Xmm Ymm Zmm' or with 'M421 In Jm Zmm'
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Minor misspelling. And the word "with" isn't used _quite this way_ in formal English. Better phrasing:

``` cpp
* Use either 'M421 X<mm> Y<mm> Z<mm>' or 'M421 I<xindex> J<yindex> Z<mm>'
```

To keep "Enter … with" you could say:

``` cpp
* Enter 'M421' with 'X<mm> Y<mm> Z<mm>' or with 'I<xindex> J<yindex> Z<mm>'
```

Sorry to be so pedantic. I don't mind merging this and just fixing it up later.

Comment (Blue-Marlin) [inline Marlin/Marlin_main.cpp]: Updated

**Pull request merged**